test/hotspot/jtreg/compiler/loopopts/superword/TestMemorySegment.java

```diff
@@ -0,0 +1,860 @@
+//  * @library /test/lib /
+//  * @run driver compiler.loopopts.superword.TestMemorySegment MixedArray
+//  */
+//
+// /*
+//  * @test id=MixedBuffer
+//  * @bug 8310190
+//  * @summary Test vectorization of loops over MemorySegment
+//  * @library /test/lib /
+//  * @run driver compiler.loopopts.superword.TestMemorySegment MixedBuffer
+//  */
+//
+// /*
+//  * @test id=mixed
+//  * @bug 8310190
+//  * @summary Test vectorization of loops over MemorySegment
+//  * @library /test/lib /
+//  * @run driver compiler.loopopts.superword.TestMemorySegment Mixed
+//  */
+
+public class TestMemorySegment {
+    public static void main(String[] args) {
+        TestFramework framework = new TestFramework(TestMemorySegmentImpl.class);
+        framework.addFlags("-DmemorySegmentProviderNameForTestVM=" + args[0]);
+        framework.setDefaultWarmup(100);
+        framework.start();
+    }
+}
+
+class TestMemorySegmentImpl {
+    static final int BACKING_SIZE = 1024 * 8;
+    static final Random RANDOM = Utils.getRandomInstance();
+
+
+    interface TestFunction {
+        Object[] run();
+    }
+
+    interface MemorySegmentProvider {
+        MemorySegment newMemorySegment();
+    }
+
+    static MemorySegmentProvider provider;
+
+    static {
+        String providerName = System.getProperty("memorySegmentProviderNameForTestVM");
+        provider = switch (providerName) {
+            case "ByteArray"        -> ( () -> { return newMemorySegmentOfByteArray(); } );
+            case "CharArray"        -> ( () -> { return newMemorySegmentOfCharArray(); } );
+            case "ShortArray"       -> ( () -> { return newMemorySegmentOfShortArray(); } );
+            case "IntArray"         -> ( () -> { return newMemorySegmentOfIntArray(); } );
+            case "LongArray"        -> ( () -> { return newMemorySegmentOfLongArray(); } );
+            case "FloatArray"       -> ( () -> { return newMemorySegmentOfFloatArray(); } );
+            case "DoubleArray"      -> ( () -> { return newMemorySegmentOfDoubleArray(); } );
+            case "ByteBuffer"       -> ( () -> { return newMemorySegmentOfByteBuffer(); } );
+            case "ByteBufferDirect" -> ( () -> { return newMemorySegmentOfByteBufferDirect(); } );
+            case "Native"           -> ( () -> { return newMemorySegmentOfNative(); } );
+            case "MixedArray"       -> ( () -> { return newMemorySegmentOfMixedArray(); } );
+            case "MixedBuffer"      -> ( () -> { return newMemorySegmentOfMixedBuffer(); } );
+            case "Mixed"            -> ( () -> { return newMemorySegmentOfMixed(); } );
+            default -> throw new RuntimeException("Test argument not recognized: " + providerName);
+        };
+    }
+
+    // List of tests
+    Map<String,TestFunction> tests = new HashMap<String,TestFunction>();
+
+    // List of gold, the results from the first run before compilation
+    Map<String,Object[]> golds = new HashMap<String,Object[]>();
```

Comment:
You can replace these with `<>`:
```suggestion
    // List of tests
    Map<String, TestFunction> tests = new HashMap<>();

    // List of gold, the results from the first run before compilation
    Map<String, Object[]> golds = new HashMap<>();
```